code-challenges.test.js

```diff
@@ -23,37 +23,97 @@ const numbersArray2 = [24, 27, 30, 33, 36]
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
-// Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+describe("capFirst", () => {
+    it("Create a function that takes in an array of words and returns an array with all the words capitalized.", () => {
+        const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+        // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+        const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+        // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+        expect(capFirst(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+        expect(capFirst(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+    })
+})    
+
+//psuedo code make function that iterates through orginal array
+// take first character using built in method charAt and touppercase it and then add the rest of the string to the end by using slice 
+//return to shoot out new array with wanted output
+
 
 // b) Create the function that makes the test pass.
 
+const capFirst = (oldArray) => {
+   return oldArray.map((string) => {
+        return(string.charAt(0)).toUpperCase() + string.slice(1)
+    }) 
+ } 
+
+ //output: PASS
+
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const vowelTester1 = "learn"
-// Expected output: 1
-const vowelTester2 = "academy"
-// Expected output: 0
-const vowelTester3 = "challenges"
-// Expected output: 2
+describe("logFirstVowel", () => { 
+    it("Create a function that takes in a string and logs the index of the first vowel.", () => {
+        const vowelTester1 = "learn"
+        // Expected output: 1
+        const vowelTester2 = "academy"
+        // Expected output: 0
+        const vowelTester3 = "challenges"
+        // Expected output: 2
+        expect(logFirstVowel(vowelTester1)).toEqual(1)
+        expect(logFirstVowel(vowelTester2)).toEqual(0)
+        expect(logFirstVowel(vowelTester3)).toEqual(2)
+    })
+})
+
 
 // b) Create the function that makes the test pass.
+//pseudo code make each index a value inside of the vowel array
+// made a value that would filter the -1 out since value showed up as -1 if there indexof does not find it (learn does not have vowels i,o,u)
+//filter through voewlArr remove numbers lesser then 0
+// use math.min to get lowest value which should be the first appearence of the vowel
+//use spread operator to expand array in order for math.min function to run
```

Comment:
Really nice work on this problem and in this section!!  I really like your pseudocode explanation on this problem as well 👏